### ISSUE TITLE
Issues with Kiosk tutorial

### ISSUE THREAD

**Issue by lurch** [raspberrypi/documentation#3184]
The power-supply section of https://www.raspberrypi.com/tutorials/how-to-use-a-raspberry-pi-in-kiosk-mode/ probably needs a bit of an update, as the "Shopping list" section says "Raspberry Pi (any model)" and "USB-C power supply unit (PSU)".
 * Models of Raspberry Pi before the Pi4 use a Micro-USB power supply
 * Pi5 is being launched with a more powerful USB-C power supply

**Comment by aallan** [comment]
There is also a display issue being reported in #2925.

**Comment by nathan-contino** [comment]
Sorry for snatching this issue, @tobybroberts -- I've been doing some standardisation work on the tutorials and I've already handled this :)

**Comment by tobybroberts** [comment]
Thank you @nathan-contino - much appreciated.

**Comment by aallan** [comment]
Seems to be folks still having issues with the tutorial. See #2925 for details.

**Comment by markostamcar** [comment]
Regarding HDMI replug issue it seems that adding "--start-maximized" to chromium-browser parameter fixes it.
Source: https://forums.raspberrypi.com/viewtopic.php?t=358317

**Comment by nathan-contino** [comment]
Already added the `--start-maximized` option to the kiosk tutorial. Is anyone aware of an option that automatically reopens crashed kiosk windows? If not, we might have to wait for Chromium to improve Bookworm support. If there's anything special happening in Bookworm besides Wayland to contribute to the frequent crashes, we might want to communicate that to the Chromium devs.

**Comment by michaelsage** [comment]
> Already added the `--start-maximized` option to the kiosk tutorial. Is anyone aware of an option that automatically reopens crashed kiosk windows? If not, we might have to wait for Chromium to improve Bookworm support. If there's anything special happening in Bookworm besides Wayland to contribute to the frequent crashes, we might want to communicate that to the Chromium devs.

Or away to create a watcher to restart chromium if it's not running

**Comment by seanosteen** [comment]
I wrote a bash script to check if chromium is still running. If it is not, the kiosk reboots. I run it under the root crontab every 15 minutes. Chromium crashes about every 3 hours and 50 minutes +/- 5 minutes on Raspberry Pi 4, 8GB, Bookworm 64 bit, Wayland:

```
#!/bin/bash
if [[ $(ps -afx | grep -c 'chromium-browser') -lt 2 ]]; then
        echo "$(date): Chrome Stopped. Rebooting Kiosk." >> /var/log/reboot.log
        /usr/sbin/reboot
fi

```

> > Already added the `--start-maximized` option to the kiosk tutorial. Is anyone aware of an option that automatically reopens crashed kiosk windows? If not, we might have to wait for Chromium to improve Bookworm support. If there's anything special happening in Bookworm besides Wayland to contribute to the frequent crashes, we might want to communicate that to the Chromium devs.
> 
> Or away to create a watcher to restart chromium if it's not running, alternatively now firefox is bundled does this work any better?

**Comment by jedahan** [comment]
Probably slightly better to use a systemd service like the one here https://raw.githubusercontent.com/arrizer/MagicMirror/487f2d1022c43ca1375ec42048f7bb0cec8db6fe/magicmirror-dashboard.service

**Comment by dansiviter** [comment]
I've just attempted to use the kiosk tutorial, and I could not get it to work using Wayland. The issues I found were:
* `~/.config/wayland.ini` does not have a section with `[autostart]` and if I add it, it never gets called (using the `1 = touch /tmp/hello` test method),
* chrome won't start with `--ozone-platform=wayland` at all. Removing it helps.

The only way I got autostart to work was via the `~/.config/autostart/*` method (as suggested [here](https://forums.raspberrypi.com/viewtopic.php?t=357535)).

**Comment by michaelsage** [comment]
I don't think that used to be in the tutorial.

**Comment by dansiviter** [comment]
I think [the tutorial](https://www.raspberrypi.com/tutorials/how-to-use-a-raspberry-pi-in-kiosk-mode/) is a little confused as it mentions both `https://realtime.nationalrail.co.uk/...` and the two URLs it's supposed to switch between `https://raspberrypi.com` and `https://time.is/London`. I think that's because something are lifted/shifted from the [train timetable tutorial](https://www.raspberrypi.com/tutorials/train-timetable-raspberry-pi-tutorial/). Either way I think they need reviewing in light of latest Bookworm build.

**Comment by ivandolman** [comment]
Maybe it is a good idea to add to the tutorial how to hide your cursor? Because for what I can find it's a real pain in the ass with bookworm.

**Comment by 1liminal1** [comment]
Any updates on this? 

The Pi 5 does not have autostart either?

**Comment by aallan** [comment]
@nathan-contino When you're back from vacation can you take a look at this and try and sort it out once and for all?

**Comment by 1liminal1** [comment]
Also, none of the old screen turn off methods seem to work with Wayland. Could we please get some direction on this? Its very frustrating not having a way to turn off the screen for those of us running kiosks as home calendar displays (not to mention just plane wasteful running the screen 24h per day) 

Stuff like this, that was working last week for some people, is now just logging people off their Pis after the latest update. 

```
wlr-randr --output HDMI-A-1 --off
wlr-randr --output HDMI-A-1 --on
```
Thanks

**Comment by aallan** [comment]
Probably some overlap with #3353.

**Comment by nathan-contino** [comment]
Not so fast!

**Comment by aallan** [comment]
Hey @nathan-contino I'm still getting alerts from the Github bot when issues go stale. Can you get that turned off? Thanks!

**Comment by nathan-contino** [comment]
@aallan just checked and you aren't in the org, team, or manually added as a collaborator these days. I suspect you just need to press the big "Unwatch" button at the top of https://github.com/raspberrypi/documentation, seems like GitHub doesn't remove your Watch status when you leave an org or team.

**Comment by aallan** [comment]
Already did that (and no it weirdly doesn't). Hmm. I'll poke around in the settings a bit more. Fairly sure the github bot is a repo level thing though, not an account level thing.

**Comment by nathan-contino** [comment]
Definitely take a look at your subscriptions (https://github.com/notifications/subscriptions), might be able to remote it on a per-issue basis there. But I'll see if I can adjust the bot settings to avoid spamming you as well!

**Comment by phieri** [comment]
The kiosk tutorial states that

> Kiosk mode works on all Raspberry Pi models.

This is confusing since the tutorial then relies on Wayfire which is (still) only available on Raspberry Pi 4 and newer.

> To achieve this, we will edit .config/wayfire.ini. .config/wayfire.ini is a configuration file used to modify the behaviour and appearance of Wayfire, which is used to render the desktop in Raspberry Pi OS.

**Comment by nathan-contino** [comment]
Great point. However, I believe we use Wayfire to render the desktop on any Raspberry Pi running the Wayland window server. It's just that Raspberry Pi 4, 400, and 5 run Wayland by default since Raspberry Pi OS Bookworm.

I've updated the tutorial to specifically call out the Wayland dependency, and added instructions on how to switch to Wayland on an older Raspberry Pi as well. I also updated our documentation with a `cmdline.txt` snippet required to switch to Wayland on those earlier models.

Hopefully these changes will make the current tutorial more accessible to folks running older models of Raspberry Pi.

Closing this issue as we've tracked down the stability issues in kiosk mode to Chrome/Wayland bugs that are both out of our control and should see improvements trickling through as more and more Linux users switch to Wayland.

I'll create a separate issue to document cursor hiding and screen turn off methods in Wayland when running Kiosk mode. Since we want the tutorials to remain short and sweet and general-purpose, I'll add that content to the documentation on Kiosk mode, which needs updating anyway.

Thanks all for the information here, everyone.

**Comment by I7T5** [comment]
Does this work for DietPi?

**Comment by lurch** [comment]
> Does this work for DietPi?

All of our [documentation](https://www.raspberrypi.com/documentation/) and [tutorials](https://www.raspberrypi.com/tutorials/) are written for Raspberry Pi OS (with a few [exceptions](https://www.raspberrypi.com/tutorials/retropie-raspberry-pi-tutorial/) ).
So if it happens to also work with third-party OSes (that are also based on Debian) then great, but if our instructions don't work on your favourite OS, then you'll have to ask on the forums of that OS for additional help.

**Comment by samskiter** [comment]
Is this still busted? I can't get a freshly flashed bookworm image (64 bit) on 4B to boot and load chromium in kiosk mode - followed the tutorial but nothing happening on boot to desktop

**Comment by LukasMendez** [comment]
I'm on the same boat as @samskiter. Followed the tutorial on my 4B, and literally nothing is happening. It just goes straight to the desktop without launching the browser or anything

**Comment by mrtruebeliever** [comment]
Problem is that RPi OS now uses wayland/labwc as default en not wayland/wayfire. The tutorial is made with wayfire in mind.
labwc uses autostart file

```
# /etc/xdg/labwc/autostart
/usr/bin/lwrespawn /usr/bin/pcmanfm --desktop --profile LXDE-pi &
/usr/bin/lwrespawn /usr/bin/wf-panel-pi &
/usr/bin/lwrespawn /usr/bin/chromium-browser https://www.raspberrypi.com --kiosk --start-maximized --fast --fast-start --no-sandbox --no-first-run --noerrdialogs --disable-translate --disable-notifications --disable-infobars --disable-pinch --disable-features=TranslateUI --disk-cache-dir=/dev/null --ozone-platform=wayland --enable-features=OverlayScrollbar --overscroll-history-navigation=0 &
/usr/bin/kanshi &
/usr/bin/lxsession-xdg-autostart
```

**Comment by lurch** [comment]
> Problem is that RPi OS now uses wayland/labwc as default en not wayland/wayfire.

Good point! I guess we need to check / update our tutorials to work with labwc instead of wayfire.
@nathan-contino perhaps this would be a good learning-task for @nllho ?

**Comment by nathan-contino** [comment]
Agreed. This could use a new set of eyes to match our new Wayland stack 😈

**Comment by sarachoagustin** [comment]
Just to add to this topic, in a clean install of Raspberry Pi OS with desktop 64bit on a Raspberry Pi 5 the file `~/.config/wayfire.ini` was missing. Creating it and adding [default content](https://github.com/raspberrypi-ui/wayfire/blob/master/wayfire.ini) does nothing. So I do [this](https://gist.github.com/seffs/2395ca640d6d8d8228a19a9995418211?permalink_comment_id=4860887#force-wayland-through-raspi-config) (go to raspi-config > Advanced > Wayland > Wayfire > reboot) That creates the file and can successfully complete the tutorial)

**Comment by DanielHabenicht** [comment]
If you want to hide the mouse cursor, it seems it's not possible via unclutter anymore. So somebody added a new way in labwc: https://github.com/labwc/labwc/pull/2633

Although I can recommend sticking to X11 until all of this is properly documented (after 4 hours of trying to get a simple kiosk up and running). 
PS: disconnecting and reconnecting a HDMI display resets the full screen mode of chrome.

**Comment by ChrisAcrobat** [comment]
> If you want to hide the mouse cursor, it seems it's not possible via unclutter anymore. So somebody added a new way in labwc: [labwc/labwc#2633](https://github.com/labwc/labwc/pull/2633) (not yet release)

@DanielHabenicht Do you know if this is released yet? I haven't been able to get to work.
